### PR TITLE
[MIPS][CallSiteInfo][NFC] Fill CallSiteInfo only when needed

### PR DESCRIPTION
**Pull request by Prabhuk**
Argument-register pairs in CallSiteInfo is only needed when EmitCallSiteInfo
is on. Currently, the pairs are always pushed to the vector but only used
when EmitCallSiteInfo is on.

Don't fill the CallSiteInfo vector unless used.

Differential Revision: https://reviews.llvm.org/D107108?id=362887

### CODE REVIEW

**Review by ilovepi — APPROVED** [review]
LGTM, but I'd wait for an owner to approve before landing.

**Comment by brad0** [comment]
cc @wzssyqa

**Comment by wzssyqa** [comment]
Will it effects any code generation?

File: llvm/lib/Target/Mips/MipsISelLowering.cpp
Comment (ilovepi) [inline llvm/lib/Target/Mips/MipsISelLowering.cpp:3384]: Actually, looking at other platforms, it seems like it’s more common to only do this when EmitCallSiteInfo is true. A comment on the previous phabricator patch mentions this. 

If MIPS requires CallSiteInfo for debugging for some reason, this can be an OR expression. 

Can you update the patch to match the other platforms here?

**Review by ilovepi — CHANGES_REQUESTED** [review]

File: llvm/lib/Target/Mips/MipsISelLowering.cpp
Comment (wzssyqa) [reply llvm/lib/Target/Mips/MipsISelLowering.cpp:3384]: Let's ask @djolertrk.

https://github.com/llvm/llvm-project/commit/170ac4be3392201d5f5e124e8a1b7d78de3f82c8
https://reviews.llvm.org/D78105

Comment (djolertrk) [reply llvm/lib/Target/Mips/MipsISelLowering.cpp:3384]: Yeah, we should check for `EmitCallSiteInfo` only here. The Debug Entry Values Feature relies on this info, but the call site information could be used for some other things as well.

Comment (Prabhuk) [reply llvm/lib/Target/Mips/MipsISelLowering.cpp]: Updated the patch as suggested! Thank you.

Comment (wzssyqa) [reply llvm/lib/Target/Mips/MipsISelLowering.cpp]: LGTM. I will accept this PR after CI finish successfully.

**Pull request merged**
